andela/mrm_api
# #164266160 Add node(level) to structure tree

#### What does this PR do?
- Add node(level) to structure tree
#### Description of Task to be completed?
- implement a method on structure model to add model
- create Graphql endpoint to add a node

#### How should this be manually tested?
- git pull and checkout branch `ft-add-node-to-structure-tree-164266160`
- run this graphql mutation to create the root node
```
mutation{
  createNode(name:"floor"){
    node{
      name
      level
      left
      right
    }
    
  }
}
```
- check the database to check if the root has been created
- note the id of the root
- run the above mutation but this time provide the parent_id which is the root id as shown below
 ```
mutation{
  createNode(name:"floor", parentId:3){
    node{
      name
      level
      left
      right
    }
    
  }
}
```
#### Any background context you want to provide?
- the parent_id of the root node is `null`
#### What are the relevant pivotal tracker stories?
[#164266160](https://www.pivotaltracker.com/story/show/164266160)
#### Screenshots
<img width="648" alt="screen shot 2019-03-05 at 17 20 53" src="https://user-images.githubusercontent.com/39129473/53811732-12bdf400-3f6b-11e9-81d7-d008faff3cc9.png">

- create node mutation and output
<img width="754" alt="screen shot 2019-03-05 at 17 09 26" src="https://user-images.githubusercontent.com/39129473/53811065-86f79800-3f69-11e9-859b-2fa64144770c.png">
